lib/pure/strutils.nim

```diff
@@ -324,6 +324,18 @@ proc toOctal*(c: char): string {.noSideEffect, rtl, extern: "nsuToOctal".} =
     result[i] = chr(val mod 8 + ord('0'))
     val = val div 8
 
+proc isEmpty*(s: string): bool {.noSideEffect, procvar, rtl, extern: "nsuIsEmpty".}=
+  ## Checks whether the string `s` is empty or not.
+  ##
+  ## In this function whitespaces are stripped from `s`
+  if s.len() == 0:
+    return true
+
+  if s == nil or s == "" or s.strip() == "":
```

Comment:
`s == nil` and `s == ""` is already covered by `s.len == 0`
`s.strip()` may be rewritten to `not s.contains(AllChars - Whitespace)` to avoid allocation. But i would rather ask if this is a common pattern for `isEmpty()` to strip spaces, because `c++::std::string::empty()` and `ruby::string::empty()` don't do that.
The whole proc may be changed to template, so it may be inlined and you will not need a separate rtl symbol for it.
All in all, i would say that `isEmpty` is too trivial to exists on its own (may be written as `str.len == 0`), but along with `isBlank` and `isPresent` (see ruby) it may look more consistent.